platform/osgi/platform-osgi-configadmin/pom.xml

```diff
@@ -0,0 +1,246 @@
+                        <Bundle-Category>osgi</Bundle-Category>
+                        <Bundle-SymbolicName>
+                            ${project.artifactId}
+                        </Bundle-SymbolicName>
+                        <Bundle-Vendor>Codice</Bundle-Vendor>
+                        <Bundle-Activator>
+                            org.apache.felix.cm.impl.ConfigurationManager
+                        </Bundle-Activator>
+                        <Export-Package>
+                            org.apache.felix.cm,
+                            org.apache.felix.cm.file,
+                            org.osgi.service.cm;provide:=true;version=1.5
+                        </Export-Package>
+                        <Import-Package>
+                            <!-- Bundle plugin will try to import this since it's a maven dependency,
+                            but we're shading the dependencies into the jar so we don't need it. Karaf
+                            won't even start if we allow this import to be included (since it's not
+                            installed in the container anymore). -->
+                            !org.apache.felix.cm.*,
+                            <!-- Temporary changes -->
+                            !org.apache.commons.*,
+                            !org.apache.kafka.*,
+                            <!--!org.apache.log4j.*,-->
+                            !com.fasterxml.*,
+                            !com.conversantmedia.*,
+                            !com.lmax.*,
+                            !org.w3c.*,
+                            !org.xml.*,
+                            !org.zeromq,
+                            !org.codehaus.*,
+                            !org.fusesource.*,
+                            !org.jctools.*,
+                            !javax.*,
+                            javax.crypto,
+                            javax.crypto.spec,
+
+                            org.osgi.service.cm;version="[1.5,1.6)",
+                            org.osgi.framework;version="[1.4,2)",
+                            org.osgi.service.log;resolution:=optional;version="1.3",
+                            *
+                        </Import-Package>
+                        <DynamicImport-Package>
+                            org.osgi.service.log;version="1.3"
+                        </DynamicImport-Package>
+                        <Provide-Capability>
+                            osgi.service;objectClass:List&lt;String&gt;="org.osgi.service.cm.ConfigurationAdmin",
+                            osgi.service;objectClass:List&lt;String&gt;="org.apache.felix.cm.PersistenceManager"
+                        </Provide-Capability>
+                        <Require-Capability>
+                            osgi.service;filter:="(objectClass=org.osgi.service.log.LogService)";effective:=active;resolution:=optional
+                        </Require-Capability>
+                        <Embed-Dependency>
```

Comment:
:question: Is it necessary to embed and shade?

Reply:
Notice that those artifacts were _excluded_ from the shading process. Only Felix was actually shaded.